### PR TITLE
gnutls: update to 3.7.3

### PR DESCRIPTION
**Pull request by cinerea0**
#### Testing the changes
- I tested the changes in this PR: **YES**

### CODE REVIEW

**Comment by cinerea0** [comment]
This includes the changes in #34268 before being closed by the author. I have been using this for 12 days at the time of writing with no problems.

**Comment by ericonr** [comment]
Any specific reason for moving to the non-stable branch (I think)?

I'm not sure what our policy here has been.

**Comment by cinerea0** [comment]
I didn't realize that gnutls followed the same versioning conventions as GNOME projects where odd minor versions aren't considered stable releases, but looking at the release notes for 3.6.16 vs 3.7.2 it definitely does. I created this PR after the old one was closed because the new version is required to update filezilla. Since that isn't a package that needs to be kept perfectly up to date I understand if that's not a good reason to ignore policy.

File: srcpkgs/gnutls/template [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Template file for 'gnutls'
 pkgname=gnutls
-version=3.6.16
+version=3.7.2
```
Comment (paper42) [inline srcpkgs/gnutls/template:3]: .3 was released

File: srcpkgs/gnutls/template [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
 revision=1
 build_style=gnu-configure
 configure_args="--with-zlib --disable-guile --disable-static
```
Comment (paper42) [inline srcpkgs/gnutls/template:6]: --with-zlib is not a configure option anymore

**Comment by cinerea0** [comment]
@paper42 This PR is for a version of gnutls that is not from the stable branch ([3.6.16 release notes say "stable"](https://lists.gnupg.org/pipermail/gnutls-help/2021-May/004707.html), [3.7.3 release notes do not](https://lists.gnupg.org/pipermail/gnutls-help/2022-January/004736.html)). Is that alright?

**Comment by paper42** [comment]
> @paper42 This PR is for a version of gnutls that is not from the stable branch ([3.6.16 release notes say "stable"](https://lists.gnupg.org/pipermail/gnutls-help/2021-May/004707.html), [3.7.3 release notes do not](https://lists.gnupg.org/pipermail/gnutls-help/2022-January/004736.html)). Is that alright?

Earlier today this was discussed on IRC in #xbps, most other distributions consider 3.7 to be stable enough including Debian and some applications now require this version, so I think we should just merge this update.

**Review by ericonr — APPROVED** [review]

**Comment by ericonr** [comment]
@paper42 or anyone else, feel free to merge. 

If @cinerea0 wants, documenting that despite being a "next" version it's what everyone uses in the commit message might be a good idea.

**Comment by ericonr** [comment]
Did it myself to speed it up. Thanks!!

**Pull request closed without merging**